relay-spans/src/span.rs

```diff
@@ -0,0 +1,447 @@
+    /// can be discarded because their keys are too long or because there are too many
+    /// attributes. If this value is 0, then no attributes were dropped.
+    #[serde(default)]
+    pub dropped_attributes_count: u32,
+    /// events is a collection of Event items.``
+    #[serde(default)]
+    pub events: Vec<Event>,
+    /// dropped_events_count is the number of dropped events. If the value is 0, then no
+    /// events were dropped.
+    #[serde(default)]
+    pub dropped_events_count: u32,
+    /// links is a collection of Links, which are references from this span to a span
+    /// in the same or different trace.
+    #[serde(default)]
+    pub links: Vec<Link>,
+    /// links is a collection of Links, which are references from this span to a span
+    /// in the same or different trace.
+    #[serde(default)]
+    pub dropped_links_count: u32,
+    /// An optional final status for this span. Semantically when Status isn't set, it means
+    /// span's status code is unset, i.e. assume STATUS_CODE_UNSET (code = 0).
+    #[serde(default)]
+    pub status: Status,
+}
+
+impl Span {
+    /// sentry_status() returns a status as defined by Sentry based on the span status.
+    pub fn sentry_status(&self) -> &'static str {
+        let status_code = self.status.code.clone();
+
+        if status_code == StatusCode::Unset || status_code == StatusCode::Ok {
+            return "ok";
+        }
+
+        if let Some(code) = self
+            .attributes
+            .clone()
+            .into_iter()
+            .find(|a| a.key == "http.status_code")
+        {
+            if let Some(code_value) = code.value.to_i64() {
+                if let Some(sentry_status) = status_codes::HTTP.get(&code_value) {
+                    return sentry_status;
+                }
+            }
+        }
+
+        if let Some(code) = self
+            .attributes
+            .clone()
+            .into_iter()
+            .find(|a| a.key == "rpc.grpc.status_code")
+        {
+            if let Some(code_value) = code.value.to_i64() {
+                if let Some(sentry_status) = status_codes::GRPC.get(&code_value) {
+                    return sentry_status;
+                }
+            }
+        }
+
+        "unknown_error"
+    }
+}
+
+impl From<Span> for EventSpan {
+    fn from(from: Span) -> Self {
+        let end_timestamp = Utc.timestamp_nanos(from.end_time_unix_nano);
+        let exclusive_time = (from.end_time_unix_nano - from.start_time_unix_nano) as f64 / 1e6f64;
+        let mut attributes: Object<Value> = Object::new();
+        let start_timestamp = Utc.timestamp_nanos(from.start_time_unix_nano);
+        for attribute in from.attributes.clone() {
+            let key: String = if let Some(key) = OTEL_TO_SENTRY_TAGS.get(attribute.key.as_str()) {
+                key.to_string()
+            } else {
+                attribute.key
+            };
+            match attribute.value {
+                AnyValue::Array(_) => todo!(),
```

Comment:
Should we fix the `todo!()`s here? We should be able to convert to `Object` or `Array`.